internal/pkg/providers/azure/azure_ipwhitelister.go

```diff
@@ -0,0 +1,187 @@
+package azure
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/network/mgmt/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+	"github.com/stakater/Whitelister/internal/pkg/utils"
+)
+
+func (a *Azure) updateSecurityRules(securityGroup network.SecurityGroup, ipPermissions []utils.IpPermission) error {
+	if a.RemoveRule {
+		err := deleteRedundantSecurityRules(a, securityGroup, ipPermissions)
```

Comment:
Method name can be improved IMO. The rules are not redundant but perhaps expired or old or extra or invalid.

Reply:
redundant means : not or no longer needed or useful; superfluous.